helpers/write_apkg.py

```
@@ -84,21 +86,55 @@ def rename_temp_file(temp_path, sanitized_name, first_deck_id):
     except OSError as e:
         if e.errno == 36:
             short_final_filename = f"deck_{uuid.uuid4().hex[:8]}.apkg"
-            final_path = os.path.join(os.getcwd(), short_final_filename)
+            # Apply the same safety checks to the fallback filename
+            short_final_filename = os.path.basename(short_final_filename)
+            final_path = os.path.join(cwd, short_final_filename)
+            
+            # Verify the fallback path is safe too
+            real_final_path = os.path.realpath(final_path)
+            if not real_final_path.startswith(real_cwd + os.sep) and not real_final_path == real_cwd:
+                raise ValueError("Path traversal attempt detected in fallback filename")
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)